src/main.rs

```diff
@@ -174,6 +177,14 @@ fn record(output_filename: &Path, pid: pid_t, sample_rate: u32, is_subcommand: b
                 }
             }
         }
+        match stop_time {
```

Comment:
idiomatic rust note: you can make this a little shorter with an `if let`:
```
if let Some(time) = stop_time {
     ...
}

Reply:
The first is terser while the second is likely nicer when there are 3 or more types to match against.